packages/ui/src/Radio.module.scss

```diff
@@ -0,0 +1,109 @@
+@use '../styles/constants' as c;
+@use '../styles/helpers' as h;
+
+.wrapper {
+  display: inline-flex;
+  // inline-flex changes vertical align compared to display: flex
+  // setting vertical-align allows us to easily use radio button
+  // inline e.g. next to the text
+  // Most common use case will be still the usage in a form where parent
+  // wrapper controls this.
+  vertical-align: bottom;
+  align-items: center;
+  cursor: pointer;
+
+  @include c.typographyBodySmall;
+  // we need to provide spacing between radios if they're in a vertical column
+  margin-bottom: c.$grid;
+
+  // hide native checkbox
+  input {
+    @include h.visuallyHidden;
+  }
+
+  // default visible checkmark
+  .checkmark {
+    width: c.$iconSizeSmall;
+    height: c.$iconSizeSmall;
+    border: c.$borderWidth solid c.$colorNeutralLight;
+    border-radius: 50%;
+    // because of ~ and + selectors used below, we need to keep the current order in DOM and
+    // fix it visually with order property
+    order: -1;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  &.inline {
+    .checkmark {
+      // in case we have inline mode, we should have a padding like native radio button has
+      margin-left: c.$grid;
+    }
+  }
+
+  & > input:checked ~ .checkmark {
+    background-color: c.$colorSuccessLight;
+    border-color: c.$colorSuccessLight;
+  }
+
+  /**
+    Hover
+   */
+  &:hover,
+  &.hover {
+    input ~ .checkmark {
+      border-color: c.$colorPrimary;
+    }
+  }
+
+  /* Style the indicator (dot/circle) */
+  .checkmark::after {
+    content: '';
+    width: 1.5 * c.$grid;
+    height: 1.5 * c.$grid;
+    border-radius: 50%;
+    background: transparent;
+  }
+
+  input:checked ~ .checkmark::after {
+    background: c.$colorNeutralWhite;
+  }
+
+  .name {
+    padding-left: c.$grid * 1.25;
+    padding-right: c.$grid * 1.25;
+  }
+
+  // Error state
+  &.error input:checked ~ .checkmark {
+    border-color: c.$colorError;
+  }
+
+  /**
+    Disabled
+  */
+  &.disabled {
+    cursor: not-allowed;
+    color: c.$colorNeutralLight;
+  }
+
+  &.focus > input:not(:disabled),
```

Comment:
nit: AFAIK, browsers do not allow focusing disabled elements, so we could safely remove `:not(:disabled)` from here.